src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,42 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+
+type PostsState = {
+  posts: Post[],
+  selectedPost: Post | null,
+  isLoaded: boolean,
+  hasError: boolean,
+};
+
+const InitialState: PostsState = {
+  posts: [],
+  selectedPost: null,
+  isLoaded: false,
+  hasError: false,
+};
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState: InitialState,
+  reducers: {
+    set: (state, action: PayloadAction<Post[]>) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
You can disable this rule for the whole file instead or add rules to your package.json file